index.js

```diff
@@ -1,55 +1,371 @@
+      const storeImgName = storeItem.name
+      makeStoreImg.setAttribute('alt', `${storeImgName}`)
+      makeStoreDiv.appendChild(makeStoreImg)
+
+    // create a button, set button text, append to li
+    const cartRemoveButton = document.createElement('button')
+    cartRemoveButton.setAttribute('class', 'store-button')
+    cartRemoveButton.setAttribute('id', `${storeImgName}`)
+    cartRemoveButton.innerText = 'Add to cart'
+    makeStoreLi.append(cartRemoveButton)
+    }
+  listenForAdd()
+}
+
+// * RENDER CART
+
+// render cart based on state cart contents
+function renderCart() {
+  console.log('called: renderCart')
+// ZERO in cart logic
+    for (let i = 0; i < stateCart.length; i++) {
+      const checkZeroQuantity = stateCart[i]
+    // if quantity in basket = 0, remove from cart
+      if (checkZeroQuantity.quantity < 1) {
+        stateCart.splice(i, 1)
+      }
+    }
+
+// clear the cart
+  selectCartUl.innerHTML = ''
+
+// loop stateCart
+  for(let i = 0; i < stateCart.length; i++) {
+  const cartItemToRender = stateCart[i]
+
+// CREATE & APPEND ELEMENTS
+  // make, edit, append cart LI 
+  const makeCartLi = document.createElement('li')
+  selectCartUl.appendChild(makeCartLi) 
+
+    // make, edit, append cart IMG to li
+    const makeCartImg = document.createElement('img')
+    makeCartImg.setAttribute('class', 'cart--item-icon')
+    cartImgSrc = cartItemToRender.id
+    makeCartImg.setAttribute('src', `assets/icons/${cartImgSrc}.svg`)
+    cartImgName = cartItemToRender.name
+    makeCartImg.setAttribute('alt', `${cartImgName}`)
+    makeCartLi.appendChild(makeCartImg)
+
+    // make, edit, append P to li
+    const makeCartP = document.createElement('p')
+    makeCartP.innerText = `${cartImgName}`
+    makeCartLi.appendChild(makeCartP)
+
+    // make, edit, append, listen - BUTTON REMOVE
+    const cartRemoveButton = document.createElement('button')
+    cartRemoveButton.setAttribute('class', 'quantity-btn remove-btn center')
+    cartRemoveButton.innerText = '-'
+    cartRemoveButton.addEventListener('click', () => {
+      // decrease quantity
+      cartItemToRender.quantity--
+      renderCart(cartItemToRender)
+    })
+    makeCartLi.append(cartRemoveButton)
+
+    // make, edit, append SPAN to li
+    const cartSpan = document.createElement('span')
+    cartSpan.setAttribute('class', 'quantity-text center')
+    cartSpan.innerHTML = `${cartItemToRender.quantity}`
+    makeCartLi.append(cartSpan)
+
+    // make, edit, append, listen - BUTTON ADD
+    const cartAddButton = document.createElement('button')
+    cartAddButton.setAttribute('class', 'quantity-btn add-btn center')
+    cartAddButton.innerText = '+'
+    cartAddButton.addEventListener('click', () => {
+      // increase quantity
+      cartItemToRender.quantity++
+      renderCart(cartItemToRender)
+    })
+    makeCartLi.append(cartAddButton)
+}
+console.log('*** UPDATED STATE CART ***', stateCart)
+cartTotal()
```

Comment:
The cart total doesn't always display to 2 digits passed the decimal point, can you think of a way to do this using the Intl.NumberFormat() function?

Reply:
Ah well spotted! I'll go back and review that!

Reply:
Added some arguments to it, now it's behaving how it should:
Intl.NumberFormat('en-UK', {style: 'currency', currency: "GBP"}).format(currentTotal)